polar-core/src/vm.rs

```diff
@@ -296,7 +296,9 @@ impl PolarVirtualMachine {
             goals: GoalStack::new_reversed(goals),
             binding_manager: BindingManager::new(),
             query_start_time: None,
-            query_timeout_millis: QUERY_TIMEOUT_MILLIS,
+            query_timeout_ms: std::env::var("POLAR_TIMEOUT_MS")
+                .map(|timeout_str| timeout_str.parse::<u64>().unwrap_or(DEFAULT_TIMEOUT_MS))
+                .unwrap_or(DEFAULT_TIMEOUT_MS),
```

Comment:
nit: slight preference for 
```suggestion
            query_timeout_ms: std::env::var("POLAR_TIMEOUT_MS")
                .ok()
                .and_then(|ms| ms.parse::<u64>().ok())
                .unwrap_or(DEFAULT_TIMEOUT_MS),
```
to avoid the double `.unwrap_or(DEFAULT_TIMEOUT_MS)`

don't have to fix if you prefer it as-is

Reply:
also might want to move this logic up above the `let mut vm = Self {` to clean up the struct initialization a bit

(we should also probably do the same for `polar_log_stderr`, but don't need to do either as part of this PR)